src/Middleware/HttpOverrides/src/IPNetwork.cs

```diff
@@ -83,21 +93,159 @@ private byte[] CreateMask()
         return mask;
     }
 
-    private static void CheckPrefixLengthRange(IPAddress prefix, int prefixLength)
+    private static bool IsValidPrefixLengthRange(IPAddress prefix, int prefixLength)
     {
         if (prefixLength < 0)
         {
-            throw new ArgumentOutOfRangeException(nameof(prefixLength));
+            return false;
         }
 
         if (prefix.AddressFamily == AddressFamily.InterNetwork && prefixLength > 32)
         {
-            throw new ArgumentOutOfRangeException(nameof(prefixLength));
+            return false;
         }
 
         if (prefix.AddressFamily == AddressFamily.InterNetworkV6 && prefixLength > 128)
```

Comment:
Unrleated to your change: but these should be `if-else if-else` to avoid double-checks, or actually I'd write it as
```c#
static bool IsValidPrefixLengthRange1(IPAddress prefix, int prefixLength)
{
    return prefix.AddressFamily switch
    {
        AddressFamily.InterNetwork => prefixLength <= 32,
        AddressFamily.InterNetworkV6 => prefixLength <= 128,
        _ => prefixLength >= 0
    };
}
```
and let Roslyn and JIT optimize it.

Reply:
Ah, that's a good point. Although with your example, the `prefixLength >= 0` check isn't be enforced for the non-default cases so negative `prefixLength` values would slip through. How about this instead?

```diff
static bool IsValidPrefixLengthRange(IPAddress prefix, int prefixLength)
{
+   if (prefixLength < 0)
+   {
+       return false;
+   }

    return prefix.AddressFamily switch
    {
        AddressFamily.InterNetwork => prefixLength <= 32,
        AddressFamily.InterNetworkV6 => prefixLength <= 128,
-       _ => prefixLength >= 0
+      _ => true
    };
}
```

Reply:
Yep, you're right 👍🏻